homeassistant/components/voice_assistant/websocket_api.py

```diff
@@ -46,38 +57,84 @@ async def websocket_run(
 
     else:
         # Construct a pipeline for the required/configured language
-        language = msg.get("language", hass.config.language)
         pipeline = Pipeline(
             name=language,
             language=language,
-            conversation_engine=None,
-            tts_engine=None,
+            stt_engine=None,  # cloud
+            conversation_engine=None,  # default agent
+            tts_engine=None,  # cloud
         )
 
-    # Run pipeline with a timeout.
-    # Events are sent over the websocket connection.
     timeout = msg.get("timeout", DEFAULT_TIMEOUT)
-    run_task = hass.async_create_task(
-        TextPipelineRequest(
-            intent_input=msg["intent_input"],
-            conversation_id=msg.get("conversation_id"),
-        ).execute(
-            PipelineRun(
-                hass,
-                connection.context(msg),
-                pipeline,
-                event_callback=lambda event: connection.send_event(
-                    msg["id"], event.as_dict()
+    ws_result: Any | None = None
+
+    intent_input = msg.get("intent_input")
+    if intent_input is None:
+        _LOGGER.debug("Running audio pipeline")
+
+        # Audio pipeline
+        audio_queue: "asyncio.Queue[bytes]" = asyncio.Queue()
+
+        async def stt_stream():
+            while chunk := await audio_queue.get():
+                _LOGGER.debug("Received %s byte(s) of audio", len(chunk))
+                yield chunk
+
+        def handle_binary(_hass, _connection, data: bytes):
+            audio_queue.put_nowait(data)
+
+        handler_id, _unregister_handler = connection.async_register_binary_handler(
+            handle_binary
+        )
+        ws_result = {"handler_id": handler_id}
+
+        run_task = hass.async_create_task(
+            AudioPipelineRequest(
```

Comment:
For a future PR we should make sure that the request input is supported for STT, Conversation and TTS.